tests/unit/frameworks/pytorch/imagenet_experiment_test.py

```diff
@@ -0,0 +1,117 @@
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see http://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
+
+import os
+import unittest
+
+import torch
+
+from nupic.research.frameworks.pytorch.imagenet import (
+    ImagenetExperiment,
+    TempFakeSavedData,
+)
+from nupic.research.frameworks.pytorch.models.resnets import resnet50
+
+
+class ImagenetExperimentTest(unittest.TestCase):
+
+    def setUp(self):
+        self.config = dict(
+
+            num_classes=10,
+
+            # Dataset path
+            data=os.path.expanduser("~"),
+
+            # Number of epochs
+            epochs=1,
+
+            # Model class. Must inherit from "torch.nn.Module"
+            model_class=resnet50,
+            # model model class arguments passed to the constructor
+            model_args=dict(),
+
+            # Optimizer class. Must inherit from "torch.optim.Optimizer"
+            optimizer_class=torch.optim.SGD,
+            # Optimizer class class arguments passed to the constructor
+            optimizer_args=dict(
+                lr=0.1,
+                weight_decay=1e-04,
+                momentum=0.9,
+                dampening=0,
+                nesterov=True
+            ),
+
+            # Learning rate scheduler class. Must inherit from "_LRScheduler"
+            lr_scheduler_class=torch.optim.lr_scheduler.StepLR,
+            # Learning rate scheduler class class arguments passed to the constructor
+            lr_scheduler_args=dict(
+                # LR decayed by 10 every 30 epochs
+                gamma=0.1,
+                step_size=30,
+            ),
+
+            # Works with progressive_resize and the available GPU memory fitting as many
+            # images as possible in each batch - dict(start_epoch: batch_size)
+            dynamic_batch_size=None,
+
+            # Loss function. See "torch.nn.functional"
+            loss_function=torch.nn.functional.cross_entropy,
+
+        )
+
+    def test_temp_data_util(self):
+
+        temp_data = TempFakeSavedData(
+            train_size=10,
+            batch_size=2,
+        )
+
+        temp_data_path = temp_data.dataset_path
+        with temp_data as data:
+            i = 0
+            for image, _ in data.train_dataloader:
+                batch_size = image.shape[0]
+                self.assertEqual(batch_size, 2)
+                i += 1
+            self.assertEqual(i, 5)  # there should be 10 / 2 batches
+
+        # Validate data was deleted.
+        data_exists = os.path.exists(temp_data_path)
+        self.assertFalse(data_exists)
+
+    def test_experiment_init(self):
+
+        exp = ImagenetExperiment()
+        temp_data = TempFakeSavedData(
+            train_size=10,
+            batch_size=2,
+            num_classes=10,
+        )
+
+        with temp_data as data:
```

Comment:
What is this testing? There's no assert here.

Reply:
It's a simple attempt to instantiate the experiment with the temp-data. I can change the name to reflect that.

Reply:
My point is more about the test structure. How would this test fail? There's no assert to check whether it succeeded or not.  For example, you could use assertRaises to see if the setup_experiment call raised an exception.

Reply:
Fair point. I'll add some assert statements for the data loaded by the experiment as that's what I hope to validate here. Please see the last commit (`ee341a6`).